sql/core/src/test/scala/org/apache/spark/sql/execution/OptimizeMetadataOnlyQuerySuite.scala

```diff
@@ -103,6 +103,20 @@ class OptimizeMetadataOnlyQuerySuite extends QueryTest with SharedSparkSession {
     "select partcol2, min(partcol1) from srcpart where partcol1 = 0 group by partcol2",
     "select max(c1) from (select partcol1 + 1 as c1 from srcpart where partcol1 = 0) t")
 
+  testMetadataOnly(
+    "SPARK-31590 The filter used by Metadata-only queries should not have Unevaluable",
+    """
+      |SELECT partcol1, MAX(partcol2) AS partcol2
+      |FROM srcpart
+      |WHERE partcol1 = (
+      |  SELECT MAX(partcol1)
+      |  FROM srcpart
+      |)
+      |AND partcol2= 'event'
+      |GROUP BY partcol1
+      |""".stripMargin
+  )
```

Comment:
I run this test on the master and it passed. Could you check this again?

Reply:
yea, I just ran it and passed.

Reply:
Sorry, I typed an extra letter t, it should be the existing partition value to filter.